components/dialog.js

```diff
@@ -0,0 +1,198 @@
+import { useEffect, useId, useState } from 'react';
+import { Dialog as HeadlessDialog } from '@headlessui/react';
+import { XIcon } from '@heroicons/react/outline';
+import { H1 } from './headings';
+import Button from './button';
+
+const EOY_MODE = true;
+
+// EOY dialog dates
+const EOY_DIALOG_START_MONTH = 12; // December
+const EOY_DIALOG_START_DAY = 4;
+const FIRST_EOY_DIALOG_END_MONTH = 12; // December
+const FIRST_EOY_DIALOG_END_DAY = 20;
+const SECOND_EOY_DIALOG_END_MONTH = 12; // December
+const SECOND_EOY_DIALOG_END_DAY = 31;
+
+// TEXT COPIES
+const FIRST_EOY_TITLE = 'You can help preserve free public access to the law.';
+const SECOND_EOY_TITLE = "Don't miss your opportunity to make an impact in 2024!";
+const FIRST_EOY_TEXT =
+  'Your monthly recurring donation helps Free Law Project bring new, innovative,' +
+  ' open-source technology to the legal ecosystem and expand free legal resources available to the public.';
+const SECOND_EOY_TEXT =
+  "Free Law Project hosts some of the world's largest open legal databases to" +
+  ' ensure that you can access, study, and improve the American legal system. We count on your support' +
+  ' for the continued growth and improved accessibility of this important legal resource.';
+
+/**
+ * Displays an End-of-Year (EOY) modal dialog that switches between two content
+ * versions based on specified date ranges within the current year.
+ *
+ * Features:
+ * - **Dismissal Logic**: When dismissed, each dialog version hides for one week.
+ * - **Customization**: Dates, titles, and texts can be adjusted via constants at the top of the file.
+ * - **Enable/Disable**: Use the `EOY_MODE` flag to toggle the dialog.
+ *
+ * Notes:
+ * - Uses `localStorage` to store dismissal timestamps; ensure this code runs on the client side.
+ * - Time comparisons are based on the user's local time zone.
+ *
+ * @returns {JSX.Element|null} The EOYDialog component or null if not displayed.
+ */
+export function EOYDialog() {
+  if (!EOY_MODE) return;
+
+  // DIALOG CLASSES
+  const dialogBaseClasses =
+    'bg-center bg-cover bg-no-repeat flex flex-col justify-between' +
+    ' items-center max-w-screen-md w-full min-h-450px max-h-screen border-purple-800';
```

Comment:
This seems about right to me, thank you, but let's remove the purple border. I think simplicity is better on that.